source/lac/sparsity_tools.cc

```diff
@@ -680,32 +687,19 @@ namespace SparsityTools
           }
 
 
-
-        // store for each coordination number the dofs with these coordination
-        // number
-        std::multimap<DynamicSparsityPattern::size_type, int>
-          dofs_by_coordination;
-
         // find coordination number for each of these dofs
+        dofs_by_coordination.clear();
         for (const types::global_dof_index next_round_dof : next_round_dofs)
-          {
-            const DynamicSparsityPattern::size_type coordination =
-              sparsity.row_length(next_round_dof);
-
-            // insert this dof at its coordination number
-            const std::pair<const DynamicSparsityPattern::size_type, int>
-              new_entry(coordination, next_round_dof);
-            dofs_by_coordination.insert(new_entry);
-          }
+          dofs_by_coordination.emplace_back(sparsity.row_length(next_round_dof),
+                                            next_round_dof);
+        std::sort(dofs_by_coordination.begin(), dofs_by_coordination.end());
 
         // assign new DoF numbers to the elements of the present front:
-        std::multimap<DynamicSparsityPattern::size_type, int>::iterator i;
-        for (i = dofs_by_coordination.begin(); i != dofs_by_coordination.end();
-             ++i)
-          new_indices[i->second] = next_free_number++;
+        for (const auto &i : dofs_by_coordination)
+          new_indices[i.second] = next_free_number++;
 
-        // after that: copy this round's dofs for the next round
-        last_round_dofs = next_round_dofs;
+        // after that: use this round's dofs for the next round
+        last_round_dofs.swap(next_round_dofs);
```

Comment:
You `clear()` `next_round_dofs` at the top of the loop, so I believe that you don't actually need what used to be in `last_round_dofs` any more at this point. If so, this would be easier to read if you did
```suggestion
        last_round_dofs = std::move(next_round_dofs);
```

Reply:
I think then I lose the already allocated memory from the round before that I would otherwise have present for `next_round_dofs`. The additional memory allocation is probably not relevant in 3d, because there are not so many rounds in the loop, but for 2d I would intuitively keep it. If you feel strongly about it, I can also adapt to your suggestion because 3d is often the real challenge.

Reply:
No, I don't feel adamant :-)